src/TestFramework/Coverage/XmlReport/PhpUnitXmlCoveredFileDataProvider.php

```diff
@@ -71,9 +70,12 @@ public function __construct(
      */
     public function provideFiles(): iterable
     {
-        $coverageIndexPath = $this->coverageDir . '/' . self::COVERAGE_INDEX_FILE_NAME;
-        $coverageIndexContent = file_get_contents($coverageIndexPath);
-
-        return $this->parser->parse($coverageIndexPath, $coverageIndexContent);
+        foreach ($this->indexParser->parse(
+            $this->indexReader->getIndexXmlPath(),
+            $this->indexReader->getIndexXmlContent()
+        ) as $infoProvider) {
+            // TODO It might be benificial to filter files at this stage, rather than later. SourceFileDataFactory does that.
```

Comment:
I'm not sure to understand this TODO

Reply:
Have a look at SourceFileDataFactory. There's another comment on this subject.